src/Microsoft.OData.Core/JsonLight/ODataJsonLightPropertyAndValueDeserializer.cs

```diff
@@ -1599,16 +3033,17 @@ private object ReadEnumValue(bool insideJsonObjectValue, IEdmEnumTypeReference e
             List<ODataProperty> properties = new List<ODataProperty>();
             while (this.JsonReader.NodeType == JsonNodeType.Property)
             {
-                this.ReadPropertyCustomAnnotationValue = this.ReadCustomInstanceAnnotationValue;
-                this.ProcessProperty(
+                this.ReadPropertyCustomAnnotationValueAsync = this.ReadCustomInstanceAnnotationValueAsync;

```

Comment:
Maybe this is not an issue, but does this value get modified inside the while loop?

Reply:
It doesn't get modified within the while loop. While I could consider moving the line outside the loop, I'm a little antsy since this logic is adapted from the sync equivalent and maybe the idea behind stashing it within the while loop was to ensure it's not assigned if there are no properties. That doesn't seem like a very compelling reason though. We'll discuss it and decide which way to go